src/main.rs

```diff
@@ -1,65 +1,66 @@
+use anyhow::{bail, ensure, Context, Result};
 use json_event_parser::{JsonEvent, JsonReader, JsonWriter};
 use std::env;
 use std::fs::File;
 use std::io::{BufRead, BufReader, Write};
 
-fn main() {
+fn main() -> Result<()> {
```

Comment:
Can you please confirm that when `main` returns `Result::Err`, the process will exit with a non-zero exit code? Did you reproduce this manually?

Reply:
Yes, I've tested this manually already.  Here's an example:

```
$ ./target/release/fil-deal-ingester
Error: Missing required argument: path to StorageMarketDeals.json.zst
$ echo $?                                                 
1
```

Reply:
Awesome, thank you!